lib/realtime_web/plugs/rls_authorization.ex

```diff
@@ -26,21 +27,32 @@ defmodule RealtimeWeb.RlsAuthorization do
          {:ok, conn} <- Authorization.get_authorizations(conn, db_conn, params) do
       conn
     else
-      error ->
-        Logger.error("Error authorizing connection: #{inspect(error)}")
-        unauthorized(conn)
+      error -> conn |> FallbackController.call(error) |> halt()
     end
   end
 
   def call(conn, _opts), do: unauthorized(conn) |> halt()
 
-  defp set_channel_params_for_authorization_check(%{path_params: path_params}, db_conn, params) do
+  defp set_channel_params_for_authorization_check(conn, db_conn, params) do
+    %{path_params: path_params, body_params: body_params} = conn
+
+    params =
+      cond do
+        conn.method == "POST" && Map.get(body_params, "name", nil) ->
+          name = Map.get(body_params, "name")
```

Comment:
Nitpick: since the default for `Map.get/2` is already `nil`, you can avoid this here. And since you don't need a default, you can use the access protocol. An additional idea is to use `Map.fetch!` in the second call, since you know it is there. WDYT?

```suggestion
        conn.method == "POST" && body_params["name"] ->
          name = Map.fetch!(body_params, "name")
```

Reply:
Thank you! I always forget that 🤦 will change 🙏